server/app/services/program/ProgramService.java

```diff
@@ -716,18 +730,30 @@ public ErrorAnd<ProgramDefinition, CiviFormError> updateLocalization(
                     .updateTranslation(locale, localizationUpdate.localizedConfirmationMessage()))
             .setStatusDefinitions(
                 programDefinition.statusDefinitions().setStatuses(toUpdateStatusesBuilder.build()));
+
     updateSummaryImageDescriptionLocalization(
         programDefinition,
         newProgram,
         localizationUpdate.localizedSummaryImageDescription(),
         locale);
 
-    return ErrorAnd.of(
-        syncProgramDefinitionQuestions(
-                programRepository.getShallowProgramDefinition(
-                    programRepository.updateProgramSync(newProgram.build().toProgram())))
-            .toCompletableFuture()
-            .join());
+    ErrorAnd<ProgramDefinition, CiviFormError> result =
+        ErrorAnd.of(
+            syncProgramDefinitionQuestions(
+                    programRepository.getShallowProgramDefinition(
+                        programRepository.updateProgramSync(newProgram.build().toProgram())))
+                .toCompletableFuture()
+                .join());
+    if (!result.isError()) {
+      // Temporary hook to create a new row in ApplicationStatuses table for the status change
```

Comment:
**suggestion:** This will only update in this case, right? Or will it sometimes create? Would you mind updating the comment if that's the case and also to also mention that this will save the statuses with its localized names set above?

Reply:
For every change in status, it will update the current active one to "obsolete" and a the latest one as active. Yes, it will do both. I will update the comment :)

Reply:
Thank you!